src/sentry/monitors/logic/mark_ok.py

```diff
@@ -43,22 +43,51 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
                 previous_checkin["status"] == CheckInStatus.OK
                 for previous_checkin in previous_checkins
             )
+        else:
+            # Mark any open incidents as recovering by default
+            incident_recovering = True
 
-            # Resolve the incident
-            if incident_recovering:
-                MonitorIncident.objects.filter(
-                    monitor_environment=monitor_env,
-                    grouphash=monitor_env.incident_grouphash,
-                ).update(
-                    resolving_checkin=checkin,
-                    resolving_timestamp=checkin.date_added,
-                )
+        # Resolve any open incidents
+        if incident_recovering:
+            # TODO(rjo100): Check for multiple open incidents due to a logic bug
```

Comment:
needed to change this logic as we were only previously resolving incidents if `recovery_threshold` was set irrespective of if `failure_issue_threshold` was set. once we drain all orphaned incidents we can change this

Reply:
Can you actually update this comment to what you wrote here in this comment haha